src/index.js

```diff
@@ -1,33 +1,23 @@
 import axios from "axios"
 import babelPolyfill from "babel-polyfill"
 import humps from "humps"
-
-class KisiError extends Error {
-    constructor(status, code = "000000", reason = null) {
-        super("An error occurred interacting with the Kisi API.")
-
-        this.status = status
-        this.code = code
-        this.reason = reason
-    }
-}
+import handleError from "./errors"
+import QueryObject from "./QueryObject"
+import globalConfig from "./config"
 
 class Kisi {
     constructor(config = {}) {
-        config = Object.assign(config, {
-            baseURL: "https://api.getkisi.com/",
-            timeout: 5000,
-            headers: {
-                "Accept": "application/json",
-                "Content-Type": "application/json",
-            }
-        })
+        config = Object.assign({}, config, globalConfig.axiosDefaultConfig)
 
         this.client = axios.create(config)
 
         this.addDecamelizationRequestInterceptor()
         this.addCamelizationResponseInterceptor()
         this.addPaginationResponseInterceptor()
+
+        for (let collection of globalConfig.supportedEndpoints){
```

Comment:
This could be done lazily, i.e. once `places` is called, it creates and returns a query object.